File: removeBOM.js
const fs = require('fs');
const path = require('path');

const folder = './3ds';

function removeBom(filePath) {
    let data = fs.readFileSync(filePath);
    if (data[0] === 0xef && data[1] === 0xbb && data[2] === 0xbf) {
        data = data.slice(3);
        fs.writeFileSync(filePath, data);
        console.log(`Removed BOM from: ${filePath}`);
    }
}

function walk(dir) {
    fs.readdirSync(dir).forEach(file => {
        const fullPath = path.join(dir, file);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            walk(fullPath);
        } else if (fullPath.endsWith('.txt')) {
            removeBom(fullPath);
        }
    });
}

walk(folder);

File: removeBOM.py
import os
from pathlib import Path

def remove_bom(file_path):
    """Remove UTF-8 BOM from file"""
    with open(file_path, 'rb') as f:
        data = f.read()

    # Check for UTF-8 BOM (0xEF 0xBB 0xBF)
    if data[:3] == b'\xef\xbb\xbf':
        data = data[3:]
        with open(file_path, 'wb') as f:
            f.write(data)
        print(f"Removed BOM from: {file_path}")

def walk(directory):
    """Recursively walk through directory and remove BOM from txt files"""
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.txt'):
                file_path = os.path.join(root, file)
                remove_bom(file_path)

if __name__ == "__main__":
    folder = Path("./switch")
    if folder.exists():
        walk(folder)
        print("✓ BOM removal completed")
    else:
        print(f"Error: {folder} does not exist")
